test/runtime/signal.c

```diff
@@ -584,20 +585,29 @@ sigsegv_handler(int signo)
 static void * 
 sigsegv_thread(void * arg)
 {
+    int *v;
     child_tid = syscall(SYS_gettid);
-    if (arg == 0) {
-        int * v = (int *)BAD_LOAD;
+    switch ((long)arg) {
+    case 0:
+        v = (int *)BAD_LOAD;
 
         /* generate sigsegv */
         *v = 1;
-    } else {
+        break;
+    case 1:
         /* should cause sigsegv as a result of general protection fault */
 #ifdef __x86_64__
         asm volatile("hlt");
 #endif
 #ifdef __aarch64__
         asm volatile("wfi");
 #endif
+        break;
+    case 2:
+        /* should cause sigsegv by passing a bad buffer address to a syscall */
+        open((const char *)BAD_LOAD, O_RDONLY);
+        assert(0); /* shouldn't arrive here... */
```

Comment:
On my mac, the open returns EFAULT instead of generating a SIGSEGV and the test fails at this assert.